### PR TITLE
adding filter functionality to redux-manifest

### PR DESCRIPTION
**Pull request by zeriley**
I'm not happy with this implementation.  It feels wrong to me, but I can't figure out why.  Show me the error of my ways.

### CODE REVIEW

File: src/util/getInMemoryPage.test.js [hunk 1]
```diff
@@ -11,4 +11,13 @@ describe('getInMemoryPage', () => {
       data: [{n: 'a'}, {n: 'B'}, {n: 'c'}]
     })
   })
+  it('applies a filterFn', () => {
+    const data = [{n: 'c'}, {n: 'B'}, {n: 'a'}]
+    const filter = {sorts: [{id: 'n', isAsc: true}], search: 'B'}
+    const filterFn = (data, search) => data.filter(datum => datum.n === search)
```
Comment (frodare) [inline src/util/getInMemoryPage.test.js]: add tests to test what happens when the filter returns `null` and `undefined`, we should log an error to speed up debugging filter bugs

Comment (Alemkin) [reply src/util/getInMemoryPage.test.js]: so picky... Whoops, how'd I get here?

Comment (zeriley) [reply src/util/getInMemoryPage.test.js]: Someone call security!
